mllib/src/main/scala/org/apache/spark/ml/feature/MaxAbsScaler.scala

```diff
@@ -39,9 +39,7 @@ private[feature] trait MaxAbsScalerParams extends Params with HasInputCol with H
 
    /** Validates and transforms the input schema. */
   protected def validateAndTransformSchema(schema: StructType): StructType = {
-    val inputType = schema($(inputCol)).dataType
-    require(inputType.isInstanceOf[VectorUDT],
-      s"Input column ${$(inputCol)} must be a vector column")
+    SchemaUtils.checkColumnType(schema, $(inputCol), new VectorUDT)
```

Comment:
just a note on this - the fact that it requires `new VectorUDT` results in a message that contains `VectorUDT@XYZ` i.e. an instance, which is ok but not ideal. For the built-in types we have a case object that makes it cleaner, so we could think about doing that for VectorUDT as it is used a lot, e.g. 
```
private[spark] case object VectorUDT extends VectorUDT
```
Or alternatively, in `SchemaUtils.checkColumnType` we could use `getClass.getName` instead.